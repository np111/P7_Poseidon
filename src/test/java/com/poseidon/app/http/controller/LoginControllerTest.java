package com.poseidon.app.http.controller;

import org.junit.jupiter.api.Test;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;

import static org.junit.jupiter.api.Assertions.*;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;

@WebMvcTest(LoginController.class)
class LoginControllerTest extends AbstractControllerTest {
    @Test
    void login() throws Exception {
        toMatchSnapshot(
                mockMvc.perform(get("/app/login")).andReturn()
        );
    }

    @Test
    void error() throws Exception {
        toMatchSnapshot(
                mockMvc.perform(get("/app/error")).andReturn()
        );
    }
}