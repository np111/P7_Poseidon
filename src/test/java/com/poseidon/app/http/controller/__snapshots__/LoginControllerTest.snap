com.poseidon.app.http.controller.LoginControllerTest.error=[
-
> GET /app/error
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title>Spring Boot</title>"
"</head>"
"<body>"
"<h3>Access Denied Exception</h3>"
"<div>"
"  Logged in user: <b class=\"user\">  </b>"
"  <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"    <input type=\"submit\" value=\"Logout\"/>"
"  </form>"
"</div>"
"<p class=\"error\">You are not authorized for the requested data.</p>"
"</body>"
"</html>   "
]


com.poseidon.app.http.controller.LoginControllerTest.login=[
-
> GET /app/login
< HTTP 302
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: http://localhost/login
""
]